castles.ts

```diff
@@ -0,0 +1,270 @@
+//% color="#AA278D" icon="\uf447"
+namespace castles {
+    /**
+     * Build a castle wall of specified width, length, and height out of blockType
+     * @param width how thick the wall will be. If at least 3, there will be a gap for inner and outer walls
+     * @param length how far a wall will span in blocks
+     * @param height the number of blocks high the wall will stand
+     * @param blockType the Minecraft block the wall will be built with
+     */
+    //% blockId=castles_build_wall
+    //% block="build castle wall|made of $blockType width $width length $length height $height" 
+    //% width.defl=3
+    //% length.defl=27
+    //% height.defl=6
+    //% blockType.defl=Block.Cobblestone
+    //% expandableArgumentMode="toggle"
+    //% blockType.fieldEditor="gridpicker"
```

Comment:
why redefine the block grid picker here instead of just setting the shadow block to be the block picker?